In [15]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyAZTpJeKTNovPTRfFJV2n_PrX-5u64uLMg'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'cuss words on twitter' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 21, 'score': {'value': 0.32764325, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.32764325, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


The toxicity of cuss words on Twitter is .32764325 which would make the result nontoxic.

In [24]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyAZTpJeKTNovPTRfFJV2n_PrX-5u64uLMg'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'cuss words on formal anti female content' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 40, 'score': {'value': 0.36095104, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.36095104, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


In [27]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyAZTpJeKTNovPTRfFJV2n_PrX-5u64uLMg'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'cuss words on formal anti male content' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 38, 'score': {'value': 0.3260917, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.3260917, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


The results above show that in formal pieces of content female and male scores are very similar and are non toxic. Anti-female content is slightly higher (score is .36095104) then anti-male scores (score is .3260917).

In [28]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyAZTpJeKTNovPTRfFJV2n_PrX-5u64uLMg'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'cuss words on informal anti male content' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 40, 'score': {'value': 0.3375374, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.3375374, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


In [34]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyAZTpJeKTNovPTRfFJV2n_PrX-5u64uLMg'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'cuss words on informal anti female content' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 42, 'score': {'value': 0.3625127, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.3625127, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


Within the informal text, both anti-male and female scores were slightly increased from the formal text. The anti-female score was still higher (.3625127) than the anti-male score (.3375374) and these scores illustrate that these scores are still not toxic.